src/UCTNode.cpp

```diff
@@ -347,6 +402,20 @@ class NodeComp : public std::binary_function<UCTNode::node_ptr_t&,
     NodeComp(int color) : m_color(color) {};
     bool operator()(const UCTNode::node_ptr_t& a,
                     const UCTNode::node_ptr_t& b) {
+
+        if (a->get_certain() || b->get_certain()) {
+            if (a->get_eval(m_color) != b->get_eval(m_color)) {
+                return a->get_eval(m_color) < b->get_eval(m_color);
+            }
+            if (a->get_eval(m_color) == 1.0 && a->get_certain() && b->get_certain()) {
+                // Reverse visit sort for certain wins, as it may lead to mate faster?
```

Comment:
Why do you think it would? It seems to have the opposite effect in my testing.
Self-play from FEN "8/3k4/8/8/8/8/3KQ3/8 w - - 0 1" (ID 141 weights, no noise)
With `cfg_use_certain=false`:
```1. Qb5+ Kd6 2. Ke3 Ke6 3. Ke4 Kd6 4. Qb4+ Kc6 5. Ke5 Kc7 6. Kd5 Kd8 7. Qb7 Ke8 8. Ke6 Kf8 9. Qf7#```
With `cfg_use_certain=true`:
```1. Qb5+ Kd6 2. Ke3 Ke6 3. Ke4 Kd6 4. Qb4+ Kc6 5. Ke5 Kc7 6. Kd5 Kd7 7. Qb7+ Kd8 8. Ke5 Ke8 9. Qg7 Kd8 10. Qb7 Ke8 11. Kd6 Kf8 12. Qh7 Ke8 13. Qg7 Kd8 14. Qa7 Ke8 15. Qg7 Kd8 16. Qa7 Ke8 17. Ke6 Kf8 18. Qh7 Ke8 19. Qh4 Kf8 20. Qg3 Ke8 21. Qg7 Kd8 22. Qb7 Ke8 23. Qb4 Kd8 24. Qc3 Ke8 25. Qh8#```
With `cfg_use_certain=true`, but without this reversal:
```1. Qb5+ Kd6 2. Ke3 Ke6 3. Ke4 Kd6 4. Qb4+ Kc6 5. Ke5 Kc7 6. Kd5 Kd7 7. Qb7+ Kd8 8. Kd6 Ke8 9. Qe7#```

Reply:
This was an unfounded guess which I hadn't tested yet.  Upon further reflection it is almost certainly always false. But as I commented separately, I've got bigger problems with this sort logic, it'll need a complete rewrite anyway.